### PR TITLE
feat: Add common response __str__ and __repr__

### PR DESCRIPTION
**Pull request by schwern**
This does most of #194. I need to move on. This can be merged as is and the rest followed up on.

* All responses have a common base class with a common `__str__` and `__repr__`.
* Error responses get their own customized version.
* Fix the signature of a GRPC response.
* Fix the ListSigningKeys docs.
* Obscure the auth_token when stringifying so it doesn't show up in logs.
* Simplify the mixins
* Remove boilerplate code from responses.
* Take advantage of dataclasses.
* Remove unnecessary dataclasses.

Review notes:
* Make sure all dataclasses have repr=False otherwise `@dataclass` provides their own and the response won't use the common `__repr__`.

What's missing is...
* Tests
* Truncating collections.

### CODE REVIEW

File: src/momento/responses/scalar_data/delete.py [hunk 1]
```diff
@@ -20,19 +17,12 @@ class CacheDeleteResponse(CacheResponse):
 class CacheDelete(ABC):
     """Groups all `CacheDeleteResponse` derived types under a common namespace."""
 
-    @dataclass
```
Comment (malandis) [inline src/momento/responses/scalar_data/delete.py:23]: dataclass also provided equals and hash methods. Do we have a substitute for that here?

**Review by malandis — COMMENTED** [review]
When on this branch and I run `client.set` I get this:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_482060/1391734938.py in <module>
----> 1 client.set(cache_name, "key", "value")

~/dev/client-sdk-python/src/momento/simple_cache_client.py in set(self, cache_name, key, value, ttl)
    269             CacheSetResponse:
    270         """
--> 271         return self._data_client.set(cache_name, key, value, ttl)
    272 
    273     def set_if_not_exists(

~/dev/client-sdk-python/src/momento/internal/synchronous/_scs_data_client.py in set(self, cache_name, key, value, ttl)
    188             execute_request_fn=execute_set_request_fn,
    189             response_fn=construct_set_response,
--> 190             error_fn=CacheSet.Error.from_sdkexception,
    191             metadata=metadata,
    192         )

AttributeError: type object 'Error' has no attribute 'from_sdkexception'
```

**Comment by schwern** [comment]
@malandis Just needed a rebase.

File: src/momento/responses/scalar_data/delete.py (hunk 1 above)
Comment (schwern) [reply src/momento/responses/scalar_data/delete.py:23]: Sounds nice, but that hasn't been a consideration for responses that I'm aware of and it isn't tested. I can put the @dataclass declarations back, I don't know that it's done consistently.

File: src/momento/responses/response.py [hunk 2]
```diff
@@ -1,5 +1,44 @@
...
+    def __repr__(self) -> str:
+        class_name = self.__class__.__qualname__
+        attributes = vars(self)
+
+        if not attributes:
+            return f"{class_name}()"
+
+        message = {k: self._truncate_value(v) for k, v in attributes.items()}
+
+        return f"{class_name}({message!r})"
```
Comment (malandis) [inline src/momento/responses/response.py:21]: repr format is keyword like arguments. With this, a `CacheGet.Hit` renders as:

`CacheGet.Hit({'value_bytes':` b'value'})` but should be `CacheGet.Hit(value_bytes=b'value')`

File: src/momento/auth/credential_provider.py [hunk 3]
```diff
@@ -65,3 +65,11 @@ def from_string(
         control_endpoint = control_endpoint or endpoints.control_endpoint
         cache_endpoint = cache_endpoint or endpoints.cache_endpoint
         return CredentialProvider(auth_token, control_endpoint, cache_endpoint)
+
+    def __repr__(self) -> str:
+        attributes: Dict[str, str] = vars(self)
+        attributes["auth_token"] = self._obscure(attributes["auth_token"])
+        return f"{self.__class__.__name__}{attributes}"
```
Comment (malandis) [inline src/momento/auth/credential_provider.py:72]: Same issue here re: repr format

File: src/momento/responses/control/signing_keys.py [hunk 4]
```diff
@@ -1,37 +1,30 @@
...
+        expires_at: datetime - when the key expires
+    """
 
-    def expires_at(self) -> datetime:
-        """Returns the datetime representation of when the key expires"""
-        return self._expires_at
+    key_id: str
+    endpoint: str
+    key: str
+    expires_at: datetime
```
[the remark this answers is not in the document]
Comment (schwern) [reply src/momento/responses/control/signing_keys.py:27]: Like this?

```
@dataclass(repr=False)
class CreateSigningKeyResponse(ControlResponse):
    """The response from creating a signing key."""

    key_id: str
    """the ID of the signing key"""
    endpoint: str
    """endpoint of the signing key"""
    key: str
    """the signing key as a JSON string"""
    expires_at: datetime
    """when the key expires"""
```

File: src/momento/responses/response.py (hunk 2 above)
Comment (schwern) [reply src/momento/responses/response.py:21]: I can't get it to do that without a lot more work.

Comment (malandis) [reply src/momento/responses/response.py]: Fixed in [67d3dea](https://github.com/momentohq/client-sdk-python/pull/258/commits/67d3dea8f1e876a5062801592254ed9fbe66c051)

File: src/momento/responses/scalar_data/delete.py (hunk 1 above)
Comment (malandis) [reply src/momento/responses/scalar_data/delete.py:23]: Added in [0ad0e4b](https://github.com/momentohq/client-sdk-python/pull/258/commits/0ad0e4b1a88e27eab579900db45f97e253211e4f) and [aae5545](https://github.com/momentohq/client-sdk-python/pull/258/commits/aae5545c6e699ea8bc0ef69f2d5b42d29e41bab3)

**Review by eaddingtonwhite — APPROVED** [review]

File: src/momento/responses/response.py [hunk 5]
```diff
@@ -1,5 +1,60 @@
...
+    # We need to work with Any.
+    @no_type_check
+    def __repr__(self) -> str:
+        class_name = self.__class__.__qualname__
+        attributes = vars(self)
+
+        if not attributes:
+            return f"{class_name}()"
+
+        message = ", ".join(f"{k}={self._truncate_value(v)!r}" for k, v in attributes.items())
```
Comment (schwern) [inline src/momento/responses/response.py:19]: Does this display bytes as `b'123'` or `"b'123'"`?

**Pull request merged**